core/templates/domain/state/StateObjectFactory.ts

```diff
@@ -181,15 +181,17 @@ export class StateObjectFactory {
       written_translations: newStateTemplate.written_translations,
       next_content_id_index: newStateTemplate.next_content_id_index
     });
-    if (newState.interaction.defaultOutcome !== null) {
+    if (newState.interaction.defaultOutcome !== null &&
+        newStateName !== null) {
       let defaultOutcome = newState.interaction.defaultOutcome;
-      defaultOutcome.dest = newStateName;
+      // Typecasting is done just to deal with TS error.
```

Comment:
Again it should be explained what TS error is fixed.

Reply:
ok